src/core/ngx_syslog.c

```diff
@@ -250,31 +258,46 @@ ngx_syslog_add_header(ngx_syslog_peer_t *peer, u_char *buf)
                        peer->hostname, &peer->tag);
 }
 
+u_char *
+ngx_syslog_add_header_rfc5424(ngx_syslog_peer_t *peer, u_char *buf)
+{
+    ngx_uint_t  pri;
+
+    pri = peer->facility * 8 + peer->severity;
+
+    if (peer->nohostname) {
+        return ngx_sprintf(buf, "<%ui>1 %V %V - nginx %d - - ", pri, &ngx_cached_syslog_rfc5424_time, ngx_log_pid);
+    }
+
+    return ngx_sprintf(buf, "<%ui>1 %V %V %V nginx %d - - ", pri, &ngx_cached_syslog_rfc5424_time, peer->hostname, ngx_log_pid);
+}
 
 void
 ngx_syslog_writer(ngx_log_t *log, ngx_uint_t level, u_char *buf,
     size_t len)
 {
-    u_char             *p, msg[NGX_SYSLOG_MAX_STR];
     ngx_uint_t          head_len;
     ngx_syslog_peer_t  *peer;
 
     peer = log->wdata;
 
+    int max_len = (peer->is_rfc5424) ? NGX_SYSLOG_RFC5424_MAX_STR : NGX_SYSLOG_MAX_STR;
+    u_char             *p, msg[max_len];
+
     if (peer->busy) {
         return;
     }
 
     peer->busy = 1;
     peer->severity = level - 1;
 
-    p = ngx_syslog_add_header(peer, msg);
+    p = (peer->is_rfc5424) ? ngx_syslog_add_header(peer, msg) : ngx_syslog_add_header_rfc5424(peer, msg);
```

Comment:
p = (peer->is_rfc5424) ? ngx_syslog_add_header_rfc5424(peer, msg) : ngx_syslog_add_header(peer, msg) ;
?

Reply:
@Appla Thanks. I fixed it